### DISCUSSION TITLE
How to change the configuration of U_MSS_BOOTMODE

### DISCUSSION THREAD

**Discussion by xiyanni** [polarfire-soc/polarfire-soc-documentation#112] [How to] [answered]
Which project(Hardware or HSS or linux or bare metal) is resposibilty for the configuration of U_MSS_BOOTMODE? 
How to change the configuration? could you help show the details? Thanks

**Comment by hughbreslin** [comment, marked as answer]
Hi @xiyanni 

The U_MSS_BOOTMODE register is programmed either through Libero or SoftConsole - our [Boot Modes Fundamentals](https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/fundamentals/boot-modes/boot-modes-fundamentals.md) documentation describes each of the boot modes and their function. 

## Changing boot mode via SoftConsole

You can change boot mode in SoftConsole using one of the "PolarFire SoC program non-secure boot mode" external tools:
![image](https://user-images.githubusercontent.com/44862881/177106734-87f47374-53fb-46cb-9f6b-a8db430deffc.png). 

To use this tool you need to build the required project, ensure it is selected in the project explorer on the left hand side and then run the tool, this will generate the required bitstream and program your device. This requires Libero or the Program and Debug tools to be installed as it relies on the `FPGENPROG` tool. 

## Changing the boot mode via Libero

There are three boot modes you can choose from in Libero: boot mode 1, boot mode 2 and boot mode 3, this can be set from the "Configure Design Initialization Data and Memories" view. You can add a boot mode 1 or 3 client to the eNVM and a boot mode 2 client to the sNVM, for example:
![image](https://user-images.githubusercontent.com/44862881/177109292-7f2d265d-7488-4a89-8758-841d4a5fd76c.png)
